cmd/xl-sets.go

```diff
@@ -1711,23 +1638,37 @@ func (s *xlSets) ListBucketsHeal(ctx context.Context) ([]BucketInfo, error) {
 
 // HealObjects - Heal all objects recursively at a specified prefix, any
 // dangling objects deleted as well automatically.
-func (s *xlSets) HealObjects(ctx context.Context, bucket, prefix string, healObjectFn func(string, string) error) (err error) {
-	recursive := true
+func (s *xlSets) HealObjects(ctx context.Context, bucket, prefix string, healObjectFn func(string, string) error) error {
 
-	endWalkCh := make(chan struct{})
-	listDir := listDirSetsFactory(ctx, s.sets...)
-	walkResultCh := startTreeWalk(ctx, bucket, prefix, "", recursive, listDir, endWalkCh)
+	marker := ""
 	for {
-		walkResult, ok := <-walkResultCh
-		if !ok {
-			break
+		if globalHTTPServer != nil {
```

Comment:
@harshavardhana, we do not need to pause healing here, this is already done in background-heal-ops.go

Reply:
Listing is slowing down itself further if needed @vadmeste - we need to comprehensively fix this by implementing suspend, resume style for these type of calls.